packages/affiliates/libs/contracts.js

```diff
@@ -27,11 +28,76 @@ function decodeAttribution(
   data,
   delimiter = "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff000000000000000000000000"
 ) {
+  assert(data, "requires data to decode");
   return data.split(delimiter)[1];
 }
 
+// Just wraps abi to pass through to contract
+// Lookup by erc20 address
+function Erc20({ abi = getAbi("ERC20"), web3 }) {
+  assert(abi, "requires abi for erc20");
+  const contract = new web3.eth.Contract(abi);
+  function decimals(address) {
+    assert(address, "requires address");
+    contract.options.address = address;
+    return contract.methods.decimals().call();
+  }
+  return {
+    decimals
+  };
+}
+
+// Wrapper for some basic emp functionality.
+// Currently we just need token and collateral info
+// Lookup by emp address
+function Emp({ abi = getAbi("ExpiringMultiParty"), web3 } = {}) {
+  assert(abi, "requires abi for expiring multi party");
+  const contract = new web3.eth.Contract(abi);
+  const erc20 = Erc20({ web3 });
+  function tokenCurrency(address) {
+    assert(address, "requires address");
+    contract.options.address = address;
+    return contract.methods.tokenCurrency().call();
+  }
+  function collateralCurrency(address) {
+    assert(address, "requires address");
+    contract.options.address = address;
+    return contract.methods.collateralCurrency().call();
+  }
+  async function tokenInfo(address) {
+    const tokenAddress = await tokenCurrency(address);
+    return {
+      address: tokenAddress,
+      decimals: await erc20.decimals(tokenAddress)
+    };
+  }
+  async function collateralInfo(address) {
+    const tokenAddress = await collateralCurrency(address);
+    return {
+      address: tokenAddress,
+      decimals: await erc20.decimals(tokenAddress)
```

Comment:
nit: since we have a lot of addresses floating around and method names that refer to other contracts, it may be a little more readable to call the `address` you pass into these methods `empAddress` just to differentiate it from other possible addresses.